### PR TITLE
Added the Logic for Inserting Virtual Players into the Squad

### PR DESCRIPTION
**Pull request by jayaramsivaramannair**
# Description

Fixes #

- What work was done? _Added the logic for inserting virtual players if the squad consists of less than 4 players_
- Why was this work done? _This was done to ensure that squad is complete and consists of 4 members (actual plus virtual)_
- What feature / user story is it for? _This addresses the user story when a squad consists of less than 4 members._
- Any relevant links or images / screenshots _Not Applicable_

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, but not tested (may need new tests)

## Has This Been Tested

- [x] No - Requires tests to be written

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by tinomen** [comment]
Looks like you have 4 space tabs setup locally. The linter is wanting 2 space tabs. EWach indentation should only be 2 spaces.

**Comment by jayaramsivaramannair** [comment]
> Looks like you have 4 space tabs setup locally. The linter is wanting 2 space tabs. EWach indentation should only be 2 spaces.

The table created as part of this commit has been removed so this should no longer be an issue.

File: api/mod/modHelpers/clusterGeneration.js [hunk 1]
```diff
@@ -41,6 +45,39 @@ const clusterGeneration = () => {
           const newMembers = await addMembers(trx, t1, t2, squad);
           // [[1], [2], [3], [4]] => [1, 2, 3, 4]
           members.push([].concat.apply([], newMembers));
+
+          //Virtual Player Code - members is an array of arrays where each each is a squad consisting of 4 team members
```
Comment (kwnie) [inline api/mod/modHelpers/clusterGeneration.js:49]: This says each each

**Review by kwnie — APPROVED** [review]
Looks great, Jayaram! Your code is easy to read and the comments help the code to be easy to understand. For your commit, "Added a hard-coded list of virtual players and the logic for inserting virtual players during creation of a squad", maybe those should be split into two commits since they are two different things.

File: api/mod/modHelpers/clusterGeneration.js (hunk 1 above)
Comment (jayaramsivaramannair) [reply api/mod/modHelpers/clusterGeneration.js:49]: Great catch! Thank you for noticing that.

**Comment by jayaramsivaramannair** [comment]
> Looks great, Jayaram! Your code is easy to read and the comments help the code to be easy to understand. For your commit, "Added a hard-coded list of virtual players and the logic for inserting virtual players during creation of a squad", maybe those should be split into two commits since they are two different things.

Thanks Kelsey. I only realized it after pushing the code.

File: data/seeds/019_Virtual_Players.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const faker = require('faker');
```
Comment (trevorjamesmartin) [inline data/seeds/019_Virtual_Players.js]: static-data > faker

we'd like to remove faker from the package requirements

**Comment by Jakobimatagi** [comment]
Code is easy to read and to follow along with using the comments. Good work! For commit "Ran npm run format to fix the formatting errors"

File: data/seeds/019_Virtual_Players.js (hunk 2 above)
Comment (jayaramsivaramannair) [reply data/seeds/019_Virtual_Players.js]: Thanks for your review. I am no longer using faker to generate the list of Virtual Players. All of the data used for list of Virtual Players is hardcoded. The line of code which you are referring to does not exist anymore. I have already deleted that. However, the seed data for children still uses faker.

**Review by jayaramsivaramannair — COMMENTED** [review]
> Code is easy to read and to follow along with using the comments. Good work! For commit "Ran npm run format to fix the formatting errors"

Thanks Jakobi

**Review by trevorjamesmartin — APPROVED** [review]
nice & clean,  good work!

File: data/seeds/019_Virtual_Players.js (hunk 2 above)
Comment (trevorjamesmartin) [reply data/seeds/019_Virtual_Players.js]: awesome!

**Pull request merged**